File: cacms_flutter/lib/core/models/appointment.dart
/// Matches backend AppointmentOut schema.
class Appointment {
  const Appointment({
    required this.appointmentId,
    required this.patientId,
    required this.doctorId,
    required this.scheduledDate,
    required this.queueNumber,
    required this.visitType,
    required this.status,
    required this.createdAt,
    required this.updatedAt,
    this.patientName,
  });

  final String appointmentId;
  final String patientId;
  final String doctorId;
  final DateTime scheduledDate;
  final int queueNumber;
  final String visitType;
  final String status;
  final DateTime createdAt;
  final DateTime updatedAt;
  final String? patientName;

  factory Appointment.fromJson(Map<String, dynamic> json) => Appointment(
        appointmentId: json['appointment_id'] as String,
        patientId: json['patient_id'] as String,
        doctorId: json['doctor_id'] as String,
        scheduledDate: DateTime.parse(json['scheduled_date'] as String),
        queueNumber: json['queue_number'] as int,
        visitType: json['visit_type'] as String,
        status: json['status'] as String,
        createdAt: DateTime.parse(json['created_at'] as String),
        updatedAt: DateTime.parse(json['updated_at'] as String),
        patientName: json['patient_name'] as String?,
      );

  Map<String, dynamic> toJson() => {
        'appointment_id': appointmentId,
        'patient_id': patientId,
        'doctor_id': doctorId,
        'scheduled_date': scheduledDate.toIso8601String(),
        'queue_number': queueNumber,
        'visit_type': visitType,
        'status': status,
        'created_at': createdAt.toIso8601String(),
        'updated_at': updatedAt.toIso8601String(),
        'patient_name': patientName,
      };
}

File: cacms/schemas/appointment.py
import uuid
from datetime import datetime, date
from typing import Optional, Literal
from pydantic import BaseModel


class AppointmentCreate(BaseModel):
    patient_id: uuid.UUID
    doctor_id: uuid.UUID
    scheduled_date: date
    visit_type: Literal["normal", "follow-up", "emergency"]


class AppointmentStatusUpdate(BaseModel):
    status: Literal["no-show", "cancelled"]


class AppointmentScheduleUpdate(BaseModel):
    scheduled_date: date


class AppointmentOut(BaseModel):
    appointment_id: uuid.UUID
    patient_id: uuid.UUID
    doctor_id: uuid.UUID
    scheduled_date: date
    queue_number: int
    visit_type: str
    status: str
    created_at: datetime
    updated_at: datetime
    # Joined fields
    patient_name: Optional[str] = None

    model_config = {"from_attributes": True}


class CallNextResult(BaseModel):
    completed_appointment_id: Optional[uuid.UUID] = None
    next_appointment_id: Optional[uuid.UUID] = None
    queue_empty: bool = False
    conflict: bool = False


class QueueDashboard(BaseModel):
    total: int
    completed: int
    remaining: int
    queue: list[AppointmentOut]
